yt_dlp/extractor/tubetugraz.py

```diff
@@ -0,0 +1,349 @@
+            lambda x: x["dcTitle"]]) or id
+
+        creator = try_get_or(episode_info, [
+            lambda x: x["mediapackage"]["creators"]["creator"],
+            lambda x: x["dcCreator"]])
+        if isinstance(creator, list):
+            creator = ", ".join(creator)
+
+        duration = try_get_or(episode_info, [
+            lambda x: x["mediapackage"]["duration"],
+            lambda x: x["dcExtent"]])
+
+        series_id = try_get_or(episode_info, [
+            lambda x: x["mediapackage"]["series"],
+            lambda x: x["dcIsPartOf"]])
+
+        series_title = try_get(
+            episode_info,
+            lambda x: x["mediapackage"]["seriestitle"]) or series_id
+
+        episode_title = title if series_title is not None else None
+
+        format_infos = try_get(
+            episode_info,
+            lambda x: x["mediapackage"]["media"]["track"]) or []
+
+        formats = []
+        format_types = defaultdict(lambda: defaultdict(int))
+        for format_info in format_infos:
+            formats.extend(self._extract_formats(format_info, format_types))
+
+        self._guess_formats(formats, format_types, id)
+        self._sort_formats(formats)
+
+        return {
+            "_type": "video",
+            "id": id,
+            "title": title,
+            "creator": creator,
+            "duration": duration,
+            "series": series_title,
+            "episode": episode_title,
+            "formats": formats
+        }
+
+    def _extract_formats(self, format_info, format_types):
+        PREFERRED_TYPE = "presentation"
+
+        url = try_get(format_info, lambda x: x["url"])
+        type = try_get(format_info, lambda x: x["type"]) or "unknown"
+        transport = try_get(format_info, lambda x: x["transport"]) or "https"
+        audio_bitrate = try_get(format_info, lambda x: x["audio"]["bitrate"])
+        video_bitrate = try_get(format_info, lambda x: x["video"]["bitrate"])
+        framerate = try_get(format_info, lambda x: x["video"]["framerate"])
+        resolution = try_get(format_info, lambda x: x["video"]["resolution"])
+
+        type = str(type).replace("/delivery", "")
+        transport = str(transport).lower()
+
+        if isinstance(audio_bitrate, int):
+            audio_bitrate = audio_bitrate / 1000
+        if isinstance(video_bitrate, int):
+            video_bitrate = video_bitrate / 1000
+        if isinstance(audio_bitrate, int) and isinstance(video_bitrate, int):
+            bitrate = audio_bitrate + video_bitrate
+        else:
+            bitrate = None
+
+        if type == PREFERRED_TYPE:
+            preference = -1
+        else:
+            preference = -2
+
+        if transport == "https":
+            formats = [{
+                "url": url,
+                "tbr": bitrate,
+                "abr": audio_bitrate,
+                "vbr": video_bitrate,
+                "framerate": framerate,
+                "resolution": resolution,
```

Comment:
instead, set height/width if possible

Reply:
thanks, set to width and height from `parse_resolution`.
in case `parse_resolution` fails, should width and height be None or unset?

currently they will be None, but I don't know if that will cause problems